### PR TITLE
lastSync value into COMPONENT_CONFIG is always updated

### LINKED ISSUE #17022: lastSync value into COMPONENT_CONFIG is always updated
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

core

### Describe the bug

It seems the [UserStorageManager](https://github.com/keycloak/keycloak/blob/15.0.2/services/src/main/java/org/keycloak/services/managers/UserStorageSyncManager.java#L149) implementation doesn't handle some "use cases" if remote federation fails (`RuntimeException` or `SynchronizationResult.ignored()` because `lastSync`  attribute into `COMPONENT_CONFIG` table will be updated anyway.

Since lastSync value has been updated with the previous federation failure, the next execution of federation will use the updated lastSync value and won't find updated/created users from previous lastSync period.

Something like the following that can manage both RuntimeException and SynchronizationResult.ignored() :

```java
@Override
public SynchronizationResult call() throws Exception {
    // See when we did last sync.
    int oldLastSync = provider.getLastSync();
    SynchronizationResult synchronizationResult = ((ImportSynchronization)factory).syncSince(Time.toDate(oldLastSync), sessionFactory, realmId, provider);
    if(!synchronizationResult.isIgnored()) {
        updateLastSyncInterval(sessionFactory, provider, realmId);
    }
    return synchronizationResult;
}
```

### Version

20.0.3

### Expected behavior

The `lastSync` should not be updated if ignored.

### Actual behavior

The `lastsync` attribute is updated and modified entries are not sync in the next iteration.

### How to Reproduce?

Just create a custom federation provider that returns ignored sometimes. The `lastSync` attribute is updated and possible modifie entries from the ignored iteration are lost.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by rmartinc**
Closes https://github.com/keycloak/keycloak/issues/17022

Just updating the `lastSync` if the result is not ignored. Following the proposal in the issue description, except that the time is taken before start the sync process to not lose possible changes generated during sync.

Little test added with a provider that always ignores any change.

### CODE REVIEW

**Review by mposolda — APPROVED** [review]
LGTM, so I am approving, but will be good to have also review from @keycloak/store-maintainers team.

My concern is, that this fix affects only old (legacy) store. I am not 100% sure if it is applicable to new store as well?

**Comment by pedroigor** [comment]
@mposolda I would just merge this one as our target store is still legacy. Is it OK for you?

**Comment by mposolda** [comment]
@pedroigor I've searched and `lastSync` looks like legacy store specific things. So +1 for your suggestion.

**Pull request merged**
